src/engine/init.c

```diff
@@ -852,6 +860,12 @@ parse(int argc, char **argv)
 		case 'I':
 			dss_instance_idx = atoi(optarg);
 			break;
+		case 'H':
+			dss_nvme_nr_hugepages = atoi(optarg);
+			break;
+		case 'S':
+			dss_nvme_hugepage_sz_kb = atoi(optarg);
+			break;
```

Comment:
'dss_nvme_mem_size' and 'dss_nvme_nr_hugepages'/'dss_nvme_hugepage_sz_kb' looks redundant to me, we'd use one or the other.

Let's just use existing dss_nvme_mem_size for this moment. The "opts.mem_size" code in bio_spdk_env_init() needs be revised, if it's obsolete in current implementation, we should just remove it. (need to check this with Tom)

Reply:
right now only mem_size can really be set on the daos server command line - number of hugepages are set in the server config and hugepage size is queried from /proc/meminfo. They are both added here so they can be passed from daos server to the engines

Reply:
I would just calculate nr_pages*page_size in control plane and pass that value to nvme_mem_size parameter to be used in the engine, sorry for duplicating the comment above